src/recv.c

```diff
@@ -32,10 +32,41 @@
 #include "internal.h"
 #include "iasync.h"
 #include "sync.h"
+#include "binary.h"
 
 #include <string.h>
 #include <stdlib.h>
 
+char * 
+mpd_recv_binary(struct mpd_connection *connection, const unsigned binary) {
+	char * data = malloc(binary);
```

Comment:
Depending on what `binary` means, this can be dangerous; for example, if the number was received from the server, this line could trick the client into allocate a hundred gigabytes of RAM.